File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        answer = s(n);
        
        return answer;
    }
    
    int s(int n){
        int answer = 0;
        
        int start = 1;
        
        while(start <= n){
            int plus = 0;
            for(int i=start;i<=n;i++){
                plus += i;
                if(plus == n){
                    answer++;
                    start++;
                    break;
                }
                if(plus > n){
                    start++;
                    break;
                }
            }
            
        }
        
        
        return answer;
    }
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    
    for i in range(1, (n+1)//2):
        result = 0
        for j in range(i, n+1):
            if result > n: break
            
            result += j
            
            if result == n:
                answer += 1
                break
    
    
    
    
    return answer + 1